### DISCUSSION TITLE
Some Infos

### DISCUSSION THREAD

**Discussion by mightymartin** [toniebox-reverse-engineering/toniebox#12] [Show and tell]
Hi, 

first at all thank you very much for your work and gathered information. Helped a lot! 

I was able to get my hands on an parts only Toniebox with ultra week wifi signal as main fault. Second fault was a corrupt FAT system on the flash. I was lucky to backup the cert, www and some files of the sys folder.

Here are the insights i got while fixing the box:

1. weak wifi signal was in my case an hardware fault of the cc3200. Swapped it an the wifi works fine.
2. the cc3200 does not boot the mcu.bin when the flash is formated and **newly** written with cc3200tool. Had to use Uniflash instead, otherwise the file security flag was missing.
3. the www folder ist mandatory if you want to have the branded network config page
4. after first boot an network config the box will perform an OTA update. Since there is no servicepack after format it try to install it an run into the same problem with newly created file (missing flag) resulting in an bootloop
5. Installing servicepack via Uniflash did the trick

Todo:
create files so the wifi SSID matches the Tonies wizard and the DHCP names fits.

Hope i could give a little bit back to this fantastic Project! And please dont mind my poor english ;)

Regards 
Martin

**Comment by SciLor** [comment]
Hey Martin,
thank you for sharing!

Did you try to flash a full flash dump instead of the file based one? I think then the service pack and the flags for the files should be okay right away.

Don't forget to change the mac of your box. There is a way by placing a file with the mac directly on the flash because the OFW Firmware uses it to communicate to the Boxine servers. The MAC is also visible in the client.der certificate iirc.

**Reply by mightymartin** [reply]
Hi,

flash dump was no option since the fat system was broken and not "bootable". Updated my UCF file to the MAC of the client.der file. (Good hint by the way)

ATM i have no problem using the box with the "wrong" MAC. OTA Works, Tonies play. Even the Kreativtonie is functional and updating.

Because i am lazy, i think i will use it until there is a problem connecting to Boxine servers, or do you see any problem in this pratice? The public/private key pair does match the box since i was able to rescue it ;) 

regards
Martin

**Reply by SciLor** [reply]
Great job, also did a manual extracting of the files in the past, just to see if this may be possible. 

A possible downside could be, that the box isn't tied to the Box-ID printed onto the cover of the box. You could easily test it if you set one of your creative tonies to play only on your own boxes. If the box is identified by the certificate, it will work fine. If it is identified by the MAC, it won't play (as the box is missing in your household). 

Besides that, you are the first confirmed case that exchanging the CC3200 works =). moritz of RevvoX tried it in the past without success.

**Reply by mightymartin** [reply]
The box is shown in my household an i am able to change the settings. Lucky me i think ;)

I asume Moritz has mounted the chip correct, so maybe my reconstruction of the filesystem from scratch did the trick. 

The cc3200 has no flash encryption, but i think there might be a bound between the chip an the flash content since it is written by the on chip bootloader. (At least the servicepack)

I suspect this, because in the production line distribution doc for the cc3xxx is no direct flash copy way mentioned.

The only way is an GANG image, which can be written to the flash directly. But this image contains basicly the files and information the cc needs to init itself.
